File: lib/models/video.dart
class Video {
  final String id;
  final String title;
  final String? videoUrl;
  final String? thumbnail;
  final int views;
  final int likes;
  final int dislikes;
  final int duration;
  final String? category;
  final String uploaderName;
  final String? uploaderAvatar;
  final String userId;
  final DateTime uploadDate;
  final String? description;

  Video({
    required this.id,
    required this.title,
    this.videoUrl,
    this.thumbnail,
    required this.views,
    required this.likes,
    required this.dislikes,
    required this.duration,
    this.category,
    required this.uploaderName,
    this.uploaderAvatar,
    required this.userId,
    required this.uploadDate,
    this.description,
  });

  factory Video.fromJson(Map<String, dynamic> json) {
    return Video(
      id: json['_id'] ?? json['id'] ?? '',
      title: json['title'] ?? 'Sin título',
      videoUrl: json['videoUrl'] ?? json['video_url'],
      thumbnail: json['thumbnailUrl'] ?? json['thumbnail'],
      views: json['views'] ?? 0,
      likes: json['likes'] ?? 0,
      dislikes: json['dislikes'] ?? 0,
      duration: json['duration'] ?? 0,
      category: json['category'],
      uploaderName: json['uploaderName'] ?? json['uploader_name'] ?? 'Usuario',
      uploaderAvatar: json['uploaderAvatar'] ?? json['uploader_avatar'],
      userId: json['userId'] ?? json['user_id'] ?? '',
      uploadDate: DateTime.tryParse(json['uploadDate'] ?? json['upload_date'] ?? '') ?? DateTime.now(),
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'videoUrl': videoUrl,
      'thumbnailUrl': thumbnail,
      'views': views,
      'likes': likes,
      'dislikes': dislikes,
      'duration': duration,
      'category': category,
      'uploaderName': uploaderName,
      'uploaderAvatar': uploaderAvatar,
      'userId': userId,
      'uploadDate': uploadDate.toIso8601String(),
      'description': description,
    };
  }
}

File: android-native/app/src/main/java/com/xneo/app/model/Video.kt
package com.xneo.app.model

import com.google.gson.annotations.SerializedName

data class Video(
    @SerializedName("id") val id: String = "",
    @SerializedName("title") val title: String = "",
    @SerializedName("videoUrl") val videoUrl: String? = null,
    @SerializedName("thumbnail") val thumbnail: String? = null,
    @SerializedName("views") val views: Int = 0,
    @SerializedName("likes") val likes: Int = 0,
    @SerializedName("dislikes") val dislikes: Int = 0,
    @SerializedName("duration") val duration: Int = 0,
    @SerializedName("category") val category: String? = null,
    @SerializedName("uploaderName") val uploaderName: String = "Usuario",
    @SerializedName("uploaderAvatar") val uploaderAvatar: String? = null,
    @SerializedName("userId") val userId: String = "",
    @SerializedName("uploadDate") val uploadDate: String = "",
    @SerializedName("description") val description: String? = null
)
